### PR TITLE
Ran clang-tidy with google-readability-casting

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1881/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/294/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/308/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/1989/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/420/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/290/

**Comment by zwoop** [comment]
Looks like there are merge conflicts now though :).

**Comment by SolidWallOfCode** [comment]
Unfortunately this makes code that fails to compile - notice the Linux build failed.
```
AIO.cc:410:69: error: static_cast from type 'volatile void*' to type 'char*' casts away qualifiers
           err = pread(a->aio_fildes, (static_cast<char *>(a->aio_buf)) + res, a->aio_nbytes - res, a->aio_offset + res);
```
And of course the compiler knows when to use `reinterpret_cast` vs. `static_cast` - it's the one that enforces the rules. I suspect it just tries them under the hood and picks the first one that it thinks will compile. Although clearly it doesn't get that quite right.

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/376/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/380/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/392/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1968/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/2075/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/505/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/638/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/2054/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/466/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/475/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/2158/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/589/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/462/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/718/

File: iocore/cache/CacheDir.cc [hunk 1]
```diff
@@ -950,11 +951,11 @@ sync_cache_dir_on_shutdown()
       }
       buflen = dirlen;
       if (ats_hugepage_enabled()) {
-        buf      = (char *)ats_alloc_hugepage(buflen);
+        buf      = static_cast<char *>(ats_alloc_hugepage(buflen));
```
Comment (jpeach) [inline iocore/cache/CacheDir.cc]: You can avoid this cast by changing `buf` into a `void *`.

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
@jpeach My thought is we use clang-tidy to convert the casts to C++ style.  Then is easier to through the the code to eliminate casts.  We cast way too much in our code.

The C++ casts do have a benefit since it does compile time checks and limits the type of casting.

**Comment by bryancall** [comment]
I will redo this PR closer to a release

**Pull request closed without merging**